ts/sagas/startup.ts

```diff
@@ -132,6 +136,17 @@ function* initializeApplicationSaga(): IterableIterator<Effect> {
   // User is autenticated, session token is valid
   //
 
+  if (maybeSessionInformation.isSome()) {
+    // the wallet token is available,
+    // proceed with starting the "watch wallet" saga
+    const pagoPaClient: PagoPaClient = PagoPaClient(pagoPaApiUrlPrefix);
+    yield fork(
```

Comment:
don't pass the token to the saga, instead create a client with that token, e.g.:

```ts
const pagoPaClient: PagoPaClient = PagoPaClient(pagoPaApiUrlPrefix, maybeSessionInformation.value.walletToken);
yield fork(
  watchWalletSaga,
  pagoPaClient
)
```

Reply:
Technically, the saga does need to know the wallet token: the pagoPA API require a different token (referred to as `pagoPaToken` in the code), that is short lived and, as such, might require refreshing during the course of actions. This refresh operation requires `walletToken` to get a new `pagoPaToken`, hence the need to pass the saga the token. 

The whole "token refresh" process could be deferred to `PagoPaClient`, but the saga would not be aware of this API-specific mechanism. Additionally, building a mocked `PagoPaClient` version (for testing purposes) and passing it to a saga that implements the refresh logic allows testing this mechanism more easily.